src/program/BlockStatement.js

```diff
@@ -213,7 +220,11 @@ export default class BlockStatement extends Node {
 	}
 
 	transpileParameters(code, transforms, indentation, introStatementGenerators) {
-		const params = this.parent.params;
+		let params = this.parent.params;
+
+		if (!params && this.parent.param) {
+			params = [this.parent.param];
+		}
```

Comment:
I'd prefer if for functions, `this.parent.params` would be passed to `transpileParameters`, and for catch clauses, `this.parent.param`.

Reply:
Sounds good, updated.